src/vs/workbench/contrib/terminal/common/terminal.ts

```diff
@@ -475,3 +475,43 @@ export const enum TERMINAL_COMMAND_ID {
 	NAVIGATION_MODE_FOCUS_NEXT = 'workbench.action.terminal.navigationModeFocusNext',
 	NAVIGATION_MODE_FOCUS_PREVIOUS = 'workbench.action.terminal.navigationModeFocusPrevious'
 }
+
+interface TerminalDataBuffer {
+	timeoutId: any;
+	data: string[];
+}
+
+export interface TerminalDataBufferer extends IDisposable {
+	onData: (e: string) => void;
+}
+
+const terminalBufferMap = new Map<number, TerminalDataBuffer>();
```

Comment:
It feels a little risky to have this as a singleton as if we want to use it again on either ext host or renderer it would lead to a hard to find bug. Could terminalBufferMap move into a class that manages the data bufferers and that way we can instantiate one on renderer and one on ext host? That way we could also write some tests for it too.

How about moving both into `common/terminalDataBuffering.ts` and then adding a `common/terminalDataBuffering.test.ts`?